sdk/storage/Azure.Storage.Blobs/src/BlobLeaseClient.cs

```diff
@@ -307,37 +410,31 @@ private void EnsureClient()
                     Response<BlobLease> response;
                     if (BlobClient != null)
                     {
-                        ResponseWithHeaders<BlobAcquireLeaseHeaders> blobClientResponse;
+                        Response rawResponse;
 
                         if (async)
                         {
-                            blobClientResponse = await BlobClient.BlobRestClient.AcquireLeaseAsync(
+                            rawResponse = await BlobClient.BlobRestClient.AcquireLeaseAsync(
                                 duration: serviceDuration,
                                 proposedLeaseId: LeaseId,
-                                ifModifiedSince: conditions?.IfModifiedSince,
-                                ifUnmodifiedSince: conditions?.IfUnmodifiedSince,
-                                ifMatch: conditions?.IfMatch?.ToString(),
-                                ifNoneMatch: conditions?.IfNoneMatch?.ToString(),
+                                requestConditions: conditions,
                                 ifTags: tagCondition,
-                                cancellationToken: cancellationToken)
+                                context: context)
                                 .ConfigureAwait(false);
                         }
                         else
                         {
-                            blobClientResponse = BlobClient.BlobRestClient.AcquireLease(
+                            rawResponse = BlobClient.BlobRestClient.AcquireLease(
                                 duration: serviceDuration,
                                 proposedLeaseId: LeaseId,
-                                ifModifiedSince: conditions?.IfModifiedSince,
-                                ifUnmodifiedSince: conditions?.IfUnmodifiedSince,
-                                ifMatch: conditions?.IfMatch?.ToString(),
-                                ifNoneMatch: conditions?.IfNoneMatch?.ToString(),
+                                requestConditions: conditions,
                                 ifTags: tagCondition,
-                                cancellationToken: cancellationToken);
+                                context: context);
                         }
 
                         response = Response.FromValue(
-                            blobClientResponse.ToBlobLease(),
-                            blobClientResponse.GetRawResponse());
+                            ResponseWithHeaders.FromValue(new BlobAcquireLeaseHeaders(rawResponse), rawResponse).ToBlobLease(),
```

Comment:
What happens here on a failed request (that I told not to throw via `RequestContext`'s classifier)?  Will creating the new headers object throw any exceptions?  If so may be we should only instantiate the object if `rawResponse.Status == 201` and use `null` otherwise?

Reply:
Alternately, we could also create two versions of `AcquireInternal` where one simply wraps the other but does the CancellationToken => RequestContext on input and Raw Response => Strong Type on output.  Then the protocol code path doesn't need to be smart enough to worry about those edge cases.

Reply:
No all these models have a lot of null safety built in and use try-gets on headers.

However, it's probably best if the internal method returns only raw response and the methods that need the response<T> do the construction instead.

Reply:
haha I did not see the second comment when I posted my response. I think we are aligned here and the changes have been made, though not in that exact structure you describe.